packages/nimbus/src/components/button/button.mdx

```diff
@@ -17,153 +17,767 @@ figmaLink: >-
+
+### Resources
+
+Deep dive on details and access design library
 
-## Basic Usage
+[Figma library](https://example.com)
 
-If unconfigured it will default to a neutral state. It is important to configure
-the Button properly to ensure it displays as intended.
+### Hierachy
+
+#### Primary actions
+
+Primary tone, filled style, medium size. For the principal call to action on the
+page. Primary buttons should only appear once per screen.
+
+- **A solid button** should be used as a primary action on a page used to
+  highlight the most important or most common action a user is likely to take in
+  that context. It serves to guide the user towards the desired path and make
+  the key interaction clear and prominent.
 
 ```jsx-live
-const App = () => <Button>I am a Button</Button>
+const App = () => (
+  <Button colorPalette="primary" size="md" variant="solid">
+   <Icons.Add />
+   Primary action
+ </Button>
+)
 ```
 
-## Events
+#### Secondary actions
+
+Primary tone, outline style, medium size. Represents a less important or
+alternative action compared to the primary button.
 
-To handle the differences between mouse-, touch- and keyboard-interactions, the
-`Button` supports `onPress*` events. The misc `onPress*` events (`onPress`,
-`onPressStart`, `onPressEnd`, etc.) provide a more seamless and inclusive user
-experience across different devices and input methods:
+- **An outlined button** should be used as for secondary actions that are less
+  important or less frequent than the primary action. They provide users with
+  alternative options without distracting from the main call to action.
 
 ```jsx-live
-const App = () => <Button tone="primary" variant="solid" onPress={()=>alert()}>I am accessible</Button>
+const App = () => (
+  <Button tone="primary" colorPalette="primary" variant="outline" size="md">
+   <Icons.Replay />
+   Secondary action
+ </Button>
+)
 ```
 
-## Custom DOM-element
+#### Tertiary actions
+
+Primary tone, link styled, xsmall size. For lesser-performed actions or a
+collection of actions — those that users are unlikely to perform.
 
-The Button component uses the native
-[HTML button element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button)
-by default, but supports rendering as a different tag to the DOM. Use the
-`asChild` property and the tag you want to render as _only_ child.
+- **Ghost buttons** are used when you want a button to be visible but less
+  visually prominent than a primary or secondary button. These can apply to
+  tertiary actions such as drop downs and contextual actions.
+- **Link buttons** can be used when the focus should be on the content rather
+  than the UI controls themselves. They are less visually distracting and allow
+  the content to take center stage. This is common in content-heavy applications
+  or dashboards. Link and Ghost styled buttons are most used in tertiary
+  actions, and are styled in X-small or 2X-small sizes.
 
 ```jsx-live
```

Comment:
The bottom margin on these is silly large, is there any way to reduce it?

Reply:
I think that is the default we can get at the moment. Theres really no styling anywhere.